.github/container/install-nccl.sh

```diff
@@ -5,31 +5,43 @@ set -ex -o pipefail
 export DEBIAN_FRONTEND=noninteractive
 export TZ=America/Los_Angeles
 
-# If NCCL is already installed, don't reinstall it. Print a message and exit
-if dpkg -s libnccl2 libnccl-dev &> /dev/null; then
-    echo "NCCL is already installed. Skipping installation."
-    exit 0
-fi
-
-apt-get update
-
 # Extract CUDA version from `nvcc --version` output line
 # Input: "Cuda compilation tools, release X.Y, VX.Y.Z"
 # Output: X.Y
 cuda_version=$(nvcc --version | sed -n 's/^.*release \([0-9]*\.[0-9]*\).*$/\1/p')
 
+# Try to get NCCL_VERSION of installed libnccl-dev
+if [[ -z $NCCL_VERSION ]]; then
+    NCCL_VERSION=$(dpkg -s libnccl-dev | sed -n "s/^Version: \(.*+cuda${cuda_version}\)$/\1/p" | head -n 1 | tr "+" "\n" | head -1)
+fi
+
+# Skip NCCL installation if both JAX_NCCL_VERSION (user defined) and
+# NCCL_VERSION (defined in nvidia/cuda containers) are unset.
+# This case means that the base container is built from a custom image with
+# a custom network communicator or unset NCCL_VERSION env variable.
+if [[ -z $JAX_NCCL_VERSION && -z $NCCL_VERSION ]]; then
+    echo "Skip NCCL installation"
+    exit 0
+fi
+
+JAX_NCCL_VERSION=${JAX_NCCL_VERSION:-$NCCL_VERSION}
+
+apt-get update
+
 # Find latest NCCL version compatible with existing CUDA by matching
 # ${cuda_version} in the package version string
-libnccl2_version=$(apt-cache show libnccl-dev | sed -n "s/^Version: \(.*+cuda${cuda_version}\)$/\1/p" | head -n 1)
-libnccl_dev_version=$(apt-cache show libnccl-dev | sed -n "s/^Version: \(.*+cuda${cuda_version}\)$/\1/p" | head -n 1)
+libnccl2_version=$(apt-cache show libnccl-dev | sed -n "s/^Version: \(${JAX_NCCL_VERSION}.*+cuda.*\)$/\1/p" | head -n 1)
+libnccl_dev_version=$(apt-cache show libnccl-dev | sed -n "s/^Version: \(${JAX_NCCL_VERSION}.*+cuda.*\)$/\1/p" | head -n 1)
 if [[ -z "${libnccl2_version}" || -z "${libnccl_dev_version}" ]]; then
     echo "Could not find compatible NCCL version for CUDA ${cuda_version}"
     exit 1
 fi
 
-apt-get install -y \
+apt-get install -y --allow-change-held-packages \
     libnccl2=${libnccl2_version} \
     libnccl-dev=${libnccl_dev_version}
 
+dpkg -s libnccl2 libnccl-dev
```

Comment:
Is this a smoke test? Is this being done because the apt-get above doesn't print the version?

Reply:
Sort of smoke test. And you're right, this is done for debugging purposes: to have all information about install packages in one place.